sympy/functions/special/spherical_harmonics.py

```diff
@@ -179,25 +179,28 @@ def fdiff(self, argindex=4):
             raise ArgumentIndexError(self, argindex)
 
     def _eval_rewrite_as_polynomial(self, n, m, theta, phi, **kwargs):
-        # TODO: Make sure n \in N
-        # TODO: Assert |m| <= n ortherwise we should return 0
-        return self.expand(func=True)
+        if n.is_integer and n in S.Naturals:
+            if Abs(m) <= n:
```

Comment:
I think that there may be an issue here that this cannot always evaluate and therefore gives an error.
Maybe something like 
```
c = is_le(Abs(m), n)
if c == True:
    return self.expand(func=True)
elif c == False:
    return S.Zero
```
In this way, if the comparison cannot be evaluated it will not break anything.